tests/Unit/Helpers/Evolution/DiscontinuousGalerkin/BoundaryCorrections.hpp

```diff
@@ -315,4 +316,211 @@ void test_boundary_correction_conservation(
+        std::string,
+        tmpl::size<typename BoundaryCorrection::dg_package_field_tags>::value>&
+        python_dg_package_data_functions,
+    const std::array<std::string, tmpl::size<VariablesTags>::value>&
+        python_dg_boundary_terms_functions,
+    const BoundaryCorrection& correction, const Mesh<FaceDim>& face_mesh,
+    const tuples::TaggedTuple<VolumeTags...>& volume_data,
+    const bool use_moving_mesh, const ::dg::Formulation dg_formulation,
+    const double epsilon, tmpl::list<FaceTags...> /*meta*/,
+    tmpl::list<DgPackageDataTags...> /*meta*/) {
+  CAPTURE(face_mesh);
+  CAPTURE(dg_formulation);
+  CAPTURE(use_moving_mesh);
+  REQUIRE(face_mesh.number_of_grid_points() >= 1);
+  using dg_package_field_tags =
+      typename BoundaryCorrection::dg_package_field_tags;
+
+  MAKE_GENERATOR(gen);
+  std::uniform_real_distribution<> dist(0.0, 1.0);
+  DataVector used_for_size{face_mesh.number_of_grid_points()};
+
+  Variables<dg_package_field_tags> package_data{used_for_size.size()};
+  const auto fields_on_face =
+      make_with_random_values<Variables<tmpl::list<FaceTags...>>>(
+          make_not_null(&gen), make_not_null(&dist), used_for_size);
+  const auto unit_normal_covector = make_with_random_values<
+      tnsr::i<DataVector, FaceDim + 1, Frame::Inertial>>(
+      make_not_null(&gen), make_not_null(&dist), used_for_size);
+  boost::optional<tnsr::I<DataVector, FaceDim + 1, Frame::Inertial>>
+      mesh_velocity{};
+  if (use_moving_mesh) {
+    mesh_velocity = make_with_random_values<
+        tnsr::I<DataVector, FaceDim + 1, Frame::Inertial>>(
+        make_not_null(&gen), make_not_null(&dist), used_for_size);
+  }
+  boost::optional<Scalar<DataVector>> normal_dot_mesh_velocity{};
+  if (static_cast<bool>(mesh_velocity)) {
+    normal_dot_mesh_velocity =
+        dot_product(*mesh_velocity, unit_normal_covector);
+  }
+
+  // Call C++ implementation of dg_package_data
+  call_dg_package_data(make_not_null(&package_data), correction, fields_on_face,
+                       volume_data, unit_normal_covector, mesh_velocity);
+
+  // Call python implementation of dg_package_data
+  size_t function_name_index = 0;
+  tmpl::for_each<dg_package_field_tags>(
+      [epsilon, &fields_on_face, &function_name_index, &mesh_velocity,
+       &normal_dot_mesh_velocity, &package_data,
+       &python_dg_package_data_functions, &python_module, &unit_normal_covector,
+       &volume_data](auto package_data_tag_v) {
+        // avoid compiler warnings if there isn't any volume data
+        (void)volume_data;
+        INFO("Testing package data");
+        using package_data_tag = tmpl::type_from<decltype(package_data_tag_v)>;
+        using ResultType = typename package_data_tag::type;
+        try {
+          CAPTURE(python_module);
+          CAPTURE(
+              gsl::at(python_dg_package_data_functions, function_name_index));
+          const auto cxx_result = pypp::call<ResultType, ConversionClassList>(
```

Comment:
seems slightly misleading to call this `cxx_result` as it's the thing that results from calling the python functions and then converting them to c++ objects.

Reply:
Agreed, should be named `python_result`!